eth/db/witness.py

```diff
@@ -0,0 +1,67 @@
+from typing import (
+    Collection,
+    Dict,
+    NamedTuple,
+    Set,
+    Tuple,
+)
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+
+from eth.abc import (
+    WitnessIndexAPI,
+)
+
+
+class AccountQueryTracker(NamedTuple):
+    did_query_bytecode: bool
+    slots_queried: Tuple[int, ...]
+
+
+class WitnessIndex(WitnessIndexAPI):
```

Comment:
Even once I knew what a `WitnessIndex` was this name kept throwing me off, it suggests that we're indexing into a collection of witnesses. Maybe one of these would work?
- I like what you were originally calling it, the `MetaWitness`
- `WitnessSketch`
- `WitnessStats`
- `WitnessKeys`
- `WitnessProtection` ;)

Reply:
Okay, maybe back to MetaWitness :+1: